app/templates/play/level/tome/spell-top-bar-view.jade

```diff
@@ -19,6 +19,10 @@ if showAmazonLogo
     .btn.btn-small.btn-illustrated.beautify-code(title=beautifyShortcutVerbose)
       .glyphicon.glyphicon-magnet
 
+  if me.canAutoFillCode()
+    button.btn.btn-small.btn-illustrated.btn-success#fill-solution
+      span(data-i18n="play_level.fill_in_solution")
```

Comment:
A screenshot of what this looks like would make review easier.

Reply:
I placed it in the CastButtonView to begin with, seeing how it behaved when there were 2 or 3 buttons, but the only way it would look ok was if the button was conditionally removed after clicking. This added complexity around checking if the source was being changed and letting the button be clicked again. 

It fits quite well at the top alongside the Hint button.